activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/tasks/TaskRuntimeVariableMappingTest.java

```diff
@@ -0,0 +1,299 @@
+        assertThat(taskVariables)
+                .isNotNull()
+                .extracting(VariableInstance::getName,
+                            VariableInstance::getValue)
+                .containsOnly(tuple("task_input_variable_name_1",
+                                    "inputmap1Value"));
+
+        Map<String, Object> variables = new HashMap<>();
+        variables.put("task_input_variable_name_1",
+                      "outputValue"); //This should not be set to 'process_variable_inputmap_1'
+        variables.put("task_output_variable_name_1",
+                      "outputTaskValue"); //This should be set to 'process_variable_outputmap_1'
+
+        completeTask(task.getId(),
+                     variables);
+
+        procVariables = processRuntime.variables(ProcessPayloadBuilder
+                                                         .variables()
+                                                         .withProcessInstanceId(process.getId())
+                                                         .build());
+        assertThat(procVariables)
+                .isNotNull()
+                .extracting(VariableInstance::getName,
+                            VariableInstance::getValue)
+                .containsOnly(
+                        tuple("process_variable_unmapped_1",
+                              "unmapped1Value"),
+                        tuple("process_variable_inputmap_1",
+                              "inputmap1Value"),
+                        tuple("process_variable_outputmap_1",
+                              "outputTaskValue")
+
+                );
+
+        processRuntime.delete(ProcessPayloadBuilder.delete().withProcessInstance(process).build());
+    }
+
+    private void completeTask(String taskId,
+                              Map<String, Object> variables) {
+
+        Task completeTask = taskRuntime.complete(TaskPayloadBuilder
+                                                         .complete()
+                                                         .withTaskId(taskId)
+                                                         .withVariables(variables)
+                                                         .build());
+        assertThat(completeTask).isNotNull();
+        assertThat(completeTask.getStatus()).isEqualTo(TaskStatus.COMPLETED);
+    }
+
+    @Test
+    public void allVariablesShouldBePassedWhenThereIsNoMapping() {
+
+        securityUtil.logInAs("garth");
+        ProcessInstance process = processRuntime.start(ProcessPayloadBuilder.start()
+                                                               .withProcessDefinitionKey(TASK_VAR_NO_MAPPING)
+                                                               .build());
+
+        Page<Task> tasks = taskRuntime.tasks(Pageable.of(0,
+                                                         50));
+
+        assertThat(tasks.getContent()).hasSize(1);
+        Task task = tasks.getContent().get(0);
```

Comment:
Block from line 163 to 167 can be extracted in a method and reused across the different tests.